.github/workflows/continuous-integration.yml

```
@@ -0,0 +1,171 @@
+# Configuration for testing and deploying with GitHub Actions
+name: CI
+
+# Only build PRs, the master branch, and tags. Pushes to branches will only be
+# built when a PR is opened.
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+
+# Use bash by default in all jobs
+defaults:
+  run:
+    # The -l {0} is necessary for conda environments to be activated
+    shell: bash -l {0}
+
+###############################################################################
+jobs:
+
+  # Check code style and linting. Run separately so this finishes fast and
+  # reports problems right away.
+  style_checks:
+    name: Style Checks
+    runs-on: ubuntu-latest
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Use the default Python (not miniconda) to run faster
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install requirements
+        run: pip install black flake8 pylint==2.4.*
+
+      - name: List installed packages
+        run: pip freeze
+
+      - name: Formatting check (black and flake8)
+        run: make check
+
+      - name: Linting (pylint)
+        run: make lint
+
+  # Run tests, build the docs, and deploy if needed
+  test:
+    name: ${{ matrix.os }} - Python ${{ matrix.python }} ${{ matrix.dependencies }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Otherwise, the workflow would stop if a single job fails. We want to
+      # run all of them to catch failures in different combinations.
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python: [3.6, 3.7, 3.8]
```

Optional: it's also possible to test against the latest 3.9 beta, something the Python release manager [strongly encourages](](https://discuss.python.org/t/python-3-8-5-released-as-a-security-hotfix-3-9-0b5-the-last-beta-before-3-9-0-also-available/4743)) maintainers of third-party Python projects to do.

```suggestion
        python: [3.6, 3.7, 3.8, 3.9-dev]
```

---

The only caveat here is that you haver to be able to build all your dependencies. Probably not a big deal for Pooch, though.

---

Is it available on conda-forge? Or should we use pip and the system Python for this one?

---

Ah, I was referring to system (GHA) Python, not sure about Conda.

---

Yeah, it might need to be a separate job using the system Python. Let's leave that for a follow-up PR